simplq/src/components/pages/Status/index.jsx

```diff
@@ -1,76 +1,47 @@
-import React, { useState, useEffect, useCallback } from 'react';
-import { TokenRequestFactory } from '../../../api/requestFactory';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useGetToken } from 'store/asyncActions';
+import { selectToken } from 'store/token';
 import styles from './status.module.scss';
 import HeaderSection from '../../common/HeaderSection';
 import StatusContainer from './StatusContainer';
 import LoadingIndicator from '../../common/LoadingIndicator';
 import StatusSidePanel from './StatusSidePanel';
 import TokenNumber from './TokenNumber';
-import { notify } from '../../../services/notification';
-import useRequest from '../../../api/useRequest';
-
-const TIMEOUT = 10000;
-let timeoutId;
 
+// TODO Rename component to token status, component and folder/page name
 function QueueStatus(props) {
   const tokenId = props.match.params.tokenId;
-  const [tokenStatusResponse, setTokenStatusResponse] = useState();
-  const [updateInProgress, setUpdateInProgress] = useState(false);
-  const { requestMaker } = useRequest();
-
-  const showNotification = useCallback(() => {
-    notify(`${tokenStatusResponse.queueName}: You've been notified by the queue manager.`);
-  }, [tokenStatusResponse]);
-
-  const oldTokenStatus = tokenStatusResponse ? tokenStatusResponse.tokenStatus : undefined;
-  const update = useCallback(() => {
-    clearTimeout(timeoutId);
-    requestMaker(TokenRequestFactory.get(tokenId)).then((response) => {
-      if (response) {
-        setTokenStatusResponse(response);
-        if (response.tokenStatus === 'NOTIFIED' && oldTokenStatus === 'WAITING') {
-          showNotification();
-        }
-      }
-      timeoutId = setTimeout(update, TIMEOUT);
-    });
-    // eslint-disable-next-line
-  }, [tokenId, oldTokenStatus]); // don't add showNotification, will result in infinite loop
+  const dispatch = useDispatch();
+  const token = useSelector(selectToken);
+  // const token = {}
+  const getToken = useGetToken();
+  const deleteToken = useGetToken();
 
   useEffect(() => {
-    update();
-    return () => clearTimeout(timeoutId);
-  }, [update]);
+    dispatch(getToken({ tokenId, refresh: true }));
+  }, [tokenId, dispatch, getToken]);
```

Comment:
`getToken()` her might cause fetch request on every render. Not 100% sure, please check, network tab, or just put some `conosle.log()` statement to see what's going on.

Reply:
Why do you think we should do here? I thought it's better to do it once inside the action than do it everytime the action than every time it is used?

Reply:
Let's merge this as is. It will make maintenance easier, and if we can polish it later.

Reply:
I've standardised for now: https://github.com/SimplQ/simplQ-frontend/pull/526/commits/bb8f29ab1564f47e8995196071731083a5c2c331

The PR will be merged on your approval.